### PR TITLE
Added a few more methods to detect number of CPU cores on device

### PR DESCRIPTION
**Pull request by lozzle**

### CODE REVIEW

**Comment by lozzle** [comment]
Reverted some weird indent problems

File: yearclass/src/main/java/com/facebook/device/yearclass/YearClass.java [hunk 1]
```diff
@@ -100,7 +100,8 @@ private static int getNumCoresYear() {
     if (cores < 1) return CLASS_UNKNOWN;
     if (cores == 1) return CLASS_2008;
     if (cores <= 3) return CLASS_2011;
-    return CLASS_2012;
+    if (cores <= 7) return CLASS_2012;
+    return CLASS_2014;
```
Comment (zlern2k) [inline yearclass/src/main/java/com/facebook/device/yearclass/YearClass.java]: Can you separate this chunk into its own diff? Changing the heuristic is scary to do without a data study.

File: yearclass/src/main/java/com/facebook/device/yearclass/DeviceInfo.java [hunk 2]
```diff
@@ -50,6 +58,39 @@ public static int getNumberOfCPUCores() {
...
+      return DEVICEINFO_UNKNOWN;
+    }
+  }
+
+  /**
+   * Converts from a CPU core information format to number of cores.
+   * @param str The CPU core information string, in the format of "0-N"
+   * @return The number of cores represented by this string
+   */
+  public static int getCoresFromFileString(String str) {
```
Comment (zlern2k) [inline yearclass/src/main/java/com/facebook/device/yearclass/DeviceInfo.java]: Change from public to /*package*/? @VisibleForTesting would be even better but probably not worth picking up the dependency.

**Comment by lozzle** [comment]
changed parsing method to protected
removed classification changes

File: yearclass/src/main/java/com/facebook/device/yearclass/DeviceInfo.java [hunk 3]
```diff
@@ -50,6 +58,39 @@ public static int getNumberOfCPUCores() {
...
+      return DEVICEINFO_UNKNOWN;
+    }
+  }
+
+  /**
+   * Converts from a CPU core information format to number of cores.
+   * @param str The CPU core information string, in the format of "0-N"
+   * @return The number of cores represented by this string
+   */
+  protected static int getCoresFromFileString(String str) {
```
Comment (zlern2k) [inline yearclass/src/main/java/com/facebook/device/yearclass/DeviceInfo.java]: "protected" is something I'm not used to seeing on a static method. I think it should just be package scoped (i.e. no qualifier).

Comment (zlern2k) [inline yearclass/src/main/java/com/facebook/device/yearclass/DeviceInfo.java]: I did pull down the patch request and verify that the unit test runs... I just don't understand why!

**Comment by lozzle** [comment]
remove protected

**Pull request merged**